crates/services/txpool_v2/src/pool.rs

```diff
@@ -609,14 +614,45 @@ where
         self.update_stats();
     }
 
-    pub fn remove_skipped_transaction(&mut self, tx_id: TxId) {
+    pub fn remove_skipped_transaction(&mut self, tx_id: TxId, reason: String) {
+        // If pre confirmation comes from the block producer node via p2p,
+        // transaction still may be inside of the TxPool.
+        // In that case first remove it and all its dependants.
+        if self.tx_id_to_storage_id.contains_key(&tx_id) {
+            let tx_status = statuses::SqueezedOut {
+                reason: Error::SkippedTransaction(format!(
+                    "Parent transaction with id: {tx_id}, was removed because of: {reason}"
+                ))
+                .to_string(),
+            };
+            self.remove_transactions_and_dependents(iter::once(tx_id), tx_status);
+        }
+
         self.extracted_outputs.new_skipped_transaction(&tx_id);
+        // TODO: need to also get contract user if the squeezed out was the contract creator.
         let coin_dependents = self.collision_manager.get_coins_spenders(&tx_id);
-        for dependent in coin_dependents {
-            let removed = self
-                .storage
-                .remove_transaction_and_dependents_subtree(dependent);
-            self.update_components_and_caches_on_removal(removed.iter());
+        if !coin_dependents.is_empty() {
+            let tx_status = statuses::SqueezedOut {
+                reason: Error::SkippedTransaction(format!(
+                    "Parent transaction with id: {tx_id}, was removed because of: {reason}"
+                ))
+                .to_string(),
+            };
+
+            for dependent in coin_dependents {
+                let removed = self
+                    .storage
+                    .remove_transaction_and_dependents_subtree(dependent);
+                self.update_components_and_caches_on_removal(removed.iter());
+                // It's not needed to inform the status manager about the skipped transaction herself
+                // because he give the information but we need to inform him about the dependents
+                // that will be deleted
+                for removed in removed {
+                    let tx_id = removed.transaction.id();
+                    self.tx_status_manager
+                        .status_update(tx_id, tx_status.clone().into());
```

Comment:
Hmm, that makes me think that maybe the status manager should grow a `statuses_update()` function which gets an iterator. It should update all received statuses at once and then call the pruning just a single time. Wdyt?